### PR TITLE
Add get_image function to IconState

### PR DESCRIPTION
**Pull request by itsmeow**
This function allows you to get a reference to the specific DynamicImage corresponding to a dir and frame.

### CODE REVIEW

File: src/icon.rs [hunk 1]
```diff
@@ -484,6 +499,49 @@ pub struct IconState {
...
+	/// Gets a specific DynamicImage from `images`, given a dir and frame.
+	pub fn get_image(&self, dir: &Dirs, frame: u32) -> Result<&DynamicImage, error::DmiError> {
+		if self.frames < frame {
+			return Err(error::DmiError::IconState(format!(
+				"Specified frame \"{}\" is larger than the number of frames ({}) in icon_state \"{}\"",
+				frame, self.frames, self.name
+			)));
+		}
+		if (self.dirs == 1 && *dir != Dirs::SOUTH)
+			|| (self.dirs == 4 && !ORDINAL_DIRS.contains(dir))
```
Comment (ZeWaka) [inline src/icon.rs]: Should the ! be there?

File: src/icon.rs [hunk 2]
```diff
@@ -484,6 +499,49 @@ pub struct IconState {
...
+			|| (self.dirs == 8 && !ALL_DIRS.contains(dir))
+		{
+			return Err(error::DmiError::IconState(format!(
```
Comment (ZeWaka) [inline src/icon.rs]: Import error DMI error as dmierror

File: src/icon.rs [hunk 3]
```diff
@@ -484,6 +499,49 @@ pub struct IconState {
 	pub unknown_settings: Option<HashMap<String, String>>,
 }
 
+impl IconState {
+	/// Gets a specific DynamicImage from `images`, given a dir and frame.
+	pub fn get_image(&self, dir: &Dirs, frame: u32) -> Result<&DynamicImage, error::DmiError> {
+		if self.frames < frame {
+			return Err(error::DmiError::IconState(format!(
+				"Specified frame \"{}\" is larger than the number of frames ({}) in icon_state \"{}\"",
```
Comment (ZeWaka) [inline src/icon.rs]: Why not byond style inline format!

File: src/icon.rs (hunk 1 above)
Comment (itsmeow) [reply src/icon.rs]: I meant cardinal dirs. Mb.

File: src/icon.rs (hunk 3 above)
Comment (itsmeow) [reply src/icon.rs]: What do you mean?

File: src/icon.rs (hunk 2 above)
Comment (itsmeow) [reply src/icon.rs]: It was already like that but ok!

**Pull request merged**